python/src/aiconfig/editor/client/src/components/SettingsPropertyRenderer.tsx

```diff
@@ -68,6 +68,18 @@ export default function SettingsPropertyRenderer({
     initialValue ?? defaultValue
   );
 
+  // If the default value (i.e. default from schema) changes, update the property value if
+  // the config value is not set. This will handle cases such as changing the prompt model
+  // selector when the model settings tab is open
```

Comment:
Perhaps an alternative solution would be to call `getState` AIConfigContext to refresh the settings (or something similar), and connect that variable as a dependency in here: https://github.com/lastmile-ai/aiconfig/blob/e88309434bba9f5ad4e271a8afb2917e8cbeb846/python/src/aiconfig/editor/client/src/components/prompt/PromptsContainer.tsx#L81

^So here instead of directly calling the callback, we can do something like:

```typescript
const prompt = getPrompt(const statePrompt = getPrompt(stateRef.current, promptId);
const settingsCallback = useCallback(() => {
  // update settings with `prompt` data instead of props.updatePromptModelSettings
  
},
[prompt]
);

```

Reply:
Sorry, I don't really follow

Reply:
prompt data comes from the reducer so it would always be stale if we aren't using onUpdatePromptModelSettings (which itself sets the reducer value)

Reply:
I guess what I mean is, this won't change the `onUpdatePromptModelSettings`, which means that the settings component won't re-render. I agree this method is more performant since we don't need to re-render the entire settings component, only the field itself if default value has changed

Reply:
> I guess what I mean is, even though prompt is using updated state, this won't change the onUpdatePromptModelSettings callback itself, which means that the settings component won't re-render.

It does re-render, since `initialValue` is the value of what's in state for the prompt (so it changes, debounced, whenever you change a setting), just would have the same 'model' propertyValue persisted.

This particular scenario is pretty complicated, though, since it's really specific to the model right now. Long-term this would be fixed by having the schemas implemented in the parsers to properly handle how to update the config.

For now, we can do this or I think we can get creative and use 'key' prop to actually remount the component (and reset it entirely to match config state) -- sort of along the lines of how I suggested we could handle undo/redo for some components. Will put up an RFC for that

Reply:
Ok, https://github.com/lastmile-ai/aiconfig/pull/1257 might be better?